server/api/shippingAddress.js

```diff
@@ -5,18 +5,28 @@ const { ShippingAddress, User } = require('../../db/models/index')
 // create shipping address
 
 router.post('/', (req, res, next) => {
-  const { name, line1, line2, city, state, zip, userId } = req.body;
-  if (!name || !line1 || !city || !state || !zip || !userId) {
-    res.status(400).send('Invalid request. Name, Address line 1, City, State, Zipcode and UserId are required fields. Please try again with all required fields.')
+  const { name, line1, line2, city, state, zip } = req.body;
+  if (!name || !line1 || !city || !state || !zip) {
+    return res.status(400).send('Invalid request. Name, Address line 1, City, State and Zipcode are required fields. Please try again with all required fields.')
   }
-  else {
-    ShippingAddress.create({ name, line1, line2, city, state, zip, userId })
+  if (req.user) {
+    ShippingAddress.create({ name, line1, line2, city, state, zip, userId: req.user.id })
       .then(address => res.status(201).send(address))
       .catch(e => {
         res.status(400).send('Error creating shipping address');
         next(e)
       })
   }
+  else {
+    ShippingAddress.create({ name, line1, line2, city, state, zip, sessionId: req.cookies.sessionId })
```

Comment:
To be safe, you might want to consider tweaking this line `sessionId: req.cookies.sessionId || 0`
I've found that sometimes the app can get into weird edge-case type states where the cookie doesn't get set, or it gets cleared, and the sessionId ends up null and breaks a lot of stuff.

Reply:
Sure will do. Btw I'm going to close this PR and add these changes to my next PR today

Reply:
Actually just kidding, merging now. That null thing will be fixed on my next PR